src/App.tsx

```diff
@@ -33,20 +34,21 @@ function App() {
   const routes = createBrowserRouter([
     {
       path: "/",
-      element: (
-        <UserLayout />
-      ),
+      element: <UserLayout />,
       errorElement: <NotFound />,
       children: [
         { index: true, element: <Landing /> },
-        { path:'explore', element: <Explore /> },
-        { path:'explore/:startData/:endDate/:persons', element: <Explore /> },
-        { path:'room-reviews', element: <Reviews /> },
-        { path:'favorite-rooms', element: <Favorites /> },
-        { path:'room-details', element: <RoomDetails /> },
-        { path:'room-details?/:startData/:endDate/:persons', element: <RoomDetails /> },
-      ]
+        { path: "explore", element: <Explore /> },
+        { path: "explore/:startData/:endDate/:persons", element: <Explore /> },
+        { path: "room-reviews", element: <Reviews /> },
+        { path: "favorite-rooms", element: <Favorites /> },
+        {
+          path: "room-details?/:startData/:endDate/:persons/:id",
```

Comment:
we can remove the start and end date and leave only the id, and store the state in the router and then in the search params.